pkg/controller/petset/pet_set_test.go

```diff
@@ -267,3 +267,32 @@ func TestPetSetBlockingPetIsCleared(t *testing.T) {
 		t.Errorf("Unexpected blocking pet, err %v: %+v", err, p)
 	}
 }
+
+func TestSyncPetSetBlockedPet(t *testing.T) {
+	psc, fc := newFakePetSetController()
+	ps := newPetSet(3)
+	i, _ := psc.syncPetSet(ps, fc.getPodList())
+	if i != len(fc.getPodList()) {
+		t.Errorf("syncPetSet should return actual amount of pods")
+	}
+}
+
+func TestPetSetReplicaCount(t *testing.T) {
+	psc, _ := newFakePetSetController()
+	// Pass testPetSetStatus function to client to test what status is being sent to API
+	psc.kubeClient = &fakeKubeClient{
```

Comment:
just do
```
psc.kubeClient = &testclient.Fake{}
```
and ditch the custom fake. Treat it like a normal client. You can even wrap it like
```

type fakePetSetClient struct {
	replicas int
	*testclient.Fake
}

func (f *fakePetSetClient) Update(petSet *apps.PetSet) (*apps.PetSet, error) {
	return f.Fake.Apps().PetSets(petSet.Namespace).Update(petSet)
}
```